### ISSUE TITLE
add_request_headers sets header value a second time if it already exists

### ISSUE THREAD

**Issue by daimoniac** [datawire/ambassador#648]
**Describe the bug**
When a request header is set in both the original request as well as via add_request_headers via the service definition, it will end up getting set twice.

**To Reproduce**

Set in service definition:
```
      add_request_headers:
        x-some-header: 'foo'
```

Make a request with the same header:
```
$ http https://mydomain.com/reflectheaders/ 'x-some-header:bar'
HTTP/1.1 200 OK
content-length: 1030
content-type: text/plain; charset=utf-8
date: Mon, 23 Jul 2018 06:45:25 GMT
server: envoy
x-envoy-upstream-service-time: 1

GET / HTTP/1.1
Host: mydomain.com
User-Agent: HTTPie/0.9.2
Accept: */*
Accept: */*
Accept-Encoding: gzip, deflate
X-Some-Header: bar
X-Some-Header: foo
X-Envoy-Expected-Rq-Timeout-Ms: 3000
X-Envoy-Original-Path: /reflectheaders/
X-Forwarded-Proto: https
X-Request-Id: cdc71154-ee4b-487c-84ac-a57cf881b861

```

**Expected behavior**
Either overwrite the original header, or don't touch it if it is already set. Would be preferrable to have both options.

**Versions (please complete the following information):**
 - Ambassador: 0.35.3
 - Kubernetes environment GKE
 - Version v1.9.6-gke.1

**Comment by ysaakpr** [comment]
Facing exactly same issue, this should be a choice to user, and should provide an option to override, or ability to remove the header in ext-auth

Unable to override the request headers using add_request_headers

Ambassador exposes a XClientSecret Header
Ext Auth, authenticat it and masks this header
And Upstream also uses the same header with different value
So I added add_request_headers, with XClientSecret: upstream secret
But I am getting the masked header
at upstream, which should be the added request header in the mapping

**Comment by kflynn** [comment]
Shipped in 0.61. :)
